scripts/prep_data_for_model.R

```diff
@@ -13,25 +53,104 @@ unique(standard_catch$subsite)
-  select(-species, -release_id)
-
-gcs_upload(standard_catch_unmarked,
+  mutate(month = month(date), # add to join with lad and yearling
+         day = day(date)) |> 
+  left_join(daily_yearling_ruleset) |> 
+  mutate(lifestage_for_model = case_when(fork_length > cutoff & !run %in% c("fall","late fall", "winter") ~ "yearling",
+                                  fork_length <= cutoff & fork_length > 45 & !run %in% c("fall","late fall", "winter") ~ "smolt",
+                                  fork_length > 45 & run %in% c("fall", "late fall", "winter", "not recorded") ~ "smolt",
+                                  fork_length > 45 & stream == "sacramento river" ~ "smolt",
+                                  fork_length <= 45 ~ "fry", # logic from flora includes week (all weeks but 7, 8, 9 had this threshold) but I am not sure this is necessary, worth talking through
+                                  T ~ NA)) |> 
+  select(-species, -release_id, -is_yearling, -month, -day, -cutoff) |> 
+  glimpse()
+
+
+# add logic to assign lifestage_for_model 
+# extrapolate lifestage for model for plus count fish/fish without fork lenghts based on weekly fl probabilities
+# Create table with prob fry, smolt, and yearlings for each stream, site, week, year
+weekly_lifestage_bins <- standard_catch_unmarked |> 
+  filter(!is.na(fork_length), count != 0) |> 
+  mutate(year = year(date), week = week(date)) |> 
+  group_by(year, week, stream, site) |> 
+  summarize(percent_fry = sum(lifestage_for_model == "fry")/n(),
+            percent_smolt = sum(lifestage_for_model == "smolt")/n(),
+            percent_yearling = sum(lifestage_for_model == "yearling")/n()) |> 
+  ungroup() |> 
+  glimpse() 
+
+# create table of all na values that need to be filled
+# 39,141 rows 
+na_filled_lifestage <- standard_catch_unmarked |> 
+  mutate(week = week(date), year = year(date)) |> 
+  filter(is.na(fork_length) & count > 0) |> 
+  left_join(weekly_lifestage_bins) |> 
+  mutate(fry = round(count * percent_fry), 
+         smolt = round(count * percent_smolt), 
+         yearling = round(count * percent_yearling)) |> 
+  select(-lifestage_for_model, -count, -week, -year) |> # remove because all na, assigning in next line
+  pivot_longer(fry:yearling, names_to = 'lifestage_for_model', values_to = 'count') |> 
+  select(-c(percent_fry, percent_smolt, percent_yearling)) |>  
+  filter(count != 0) |> # remove 0 values introduced when 0 prop of a lifestage, significantly decreases size of DF 
+  mutate(model_lifestage_method = "assign count based on weekly distribution") |> 
+  glimpse()
+
+# add filled values back into combined_rst 
+# first filter combined rst to exclude rows in na_to_fill
+# total of 
+combined_rst_wo_na_fl <- standard_catch_unmarked |>   
+  filter(!is.na(fork_length)) |> 
+  mutate(model_lifestage_method = "assigned from fl cutoffs") |> 
+  glimpse()
+
```

Comment:
I added this no catch part because i think we want to retain the days where catch was 0.